packages/api/src/timeseries/timeseries.ts

```diff
@@ -67,17 +67,16 @@ export interface TimeSeriesPoint<T extends string | number> {
   value: T;
 }
 
-export class TimeSeriesProperty<T extends number | string> {
-  constructor(
-    /**
-     * Queries the first point of the Timeseries
-     */
-    public getFirstPoint: () => Promise<TimeSeriesPoint<T>>,
-    /**
-     * Queries the last point of the Timeseries
-     */
-    public getLastPoint: () => Promise<TimeSeriesPoint<T>>,
-    /**
+export interface TimeSeriesProperty<T extends number | string> {
```

Comment:
Why did we switch this back to an interface? I remember you mentioning that we wanted to make this a class since its slightly more performant?

Reply:
Yea, I should have been more clear and I should have looked more closely at the revised PR.

I want the consumer to have an interface but I don't want them constructing it. The class, since it just took the functions and assigned them to itself was no better than a raw object the way it was. The way it is now, we aren't constructing additional function objects per call, nor additional scopes that must be kept by the engine.